src/Microsoft.ML.CpuMath/AvxIntrinsics.cs

```diff
@@ -141,6 +141,32 @@ private static Vector256<float> GetNewDst256(in Vector256<float> xDst1, in Vecto
             return Avx.And(Avx.Subtract(xDst1, x2), xCond);
         }
 
+        [MethodImplAttribute(MethodImplOptions.AggressiveInlining)]
+        private static unsafe Vector256<float> MultiplyAdd(float* psrc1, Vector256<float> src2, Vector256<float> src3)
+        {
+            if (Fma.IsSupported)
+            {
+                return Fma.MultiplyAdd(Avx.LoadVector256(psrc1), src2, src3);
+            }
+            else
+            {
+                return Avx.Add(Avx.Multiply(Avx.LoadVector256(psrc1), src2), src3);
```

Comment:
nit: this might be more readable (and I believe the codegen should be the same) if we do:
```csharp
Vector256<float> product = Avx.Multiply(src2, Avx.LoadVector256(pSrc1));
return Avx.Add(product, src3);
```